src/zarr/core/common.py

```diff
@@ -39,6 +40,7 @@
 JSON = str | int | float | Mapping[str, "JSON"] | Sequence["JSON"] | None
 MemoryOrder = Literal["C", "F"]
 AccessModeLiteral = Literal["r", "r+", "a", "w", "w-"]
+ANY_ACCESS_MODE: Final = "r", "r+", "a", "w", "w-"
```

Comment:
```suggestion
ANY_ACCESS_MODE: Final = get_args(AccessModeLiteral)
```
It might be possible to de-duplicate here (untested) using [`typing.get_args`](https://docs.python.org/3/library/typing.html#typing.get_args)?

Reply:
at least on my machine, the type inference machinery thinks `get_args(...)` returns `tuple[Any,...]`. So this won't work.

Reply:
it does matter, because when you annotate a tuple of strings as `Final`, type checkers will know that it's `Literal[*stuff_in_the_tuple]`, which means you can use that variable for "compile-time" and runtime type checks.

Reply:
The CI passes. @d-v-b are you getting a type error or runtime error?

Reply:
> yeah it's a handful strings, so writing it down twice in essentially the exact same location is not expensive. If we are really worried that the type and the final variable might diverge, we can test for that (using get_args).

this seems like I reasonable compromise. I added a test in [`202d606` (#3156)](https://github.com/zarr-developers/zarr-python/pull/3156/commits/202d60689fe900537f40fcced8365edf41793805).